lib/fluent/plugin/in_tail.rb

```diff
@@ -502,16 +515,14 @@ def update_watcher(tail_watcher, pe, new_inode)
       new_target_info = TargetInfo.new(path, new_inode)
 
       if @follow_inodes
-        # When follow_inodes is true, it's not cleaned up by refresh_watcher.
-        # So it should be unwatched here explicitly.
-        tail_watcher.unwatched = true
```

Comment:
If possible, could you let me know who will call `detach_watcher()` if we removed this line?

I agree that `update_watcher()` shouldn't do `tail_watcher.unwatched = true` when fluentd still needs to watch the rotated file.
But how about after fluentd stopped watching the rotated file?

In my understanding, `refresh_watchers()` will never call stop_watchers() for the rotated file since `update_watcher()` has already removed its TailWatcher from `@tails`.
Fluentd will never reset the previous position to zero in pos file if nobody detachs.
So I'm worried if fluentd reads the previous position of the old rotated file by mistake when start monitoring a new file which has the same inode.
https://github.com/fluent/fluentd/blob/ce898dd572d5745663abd456486169d772daa8aa/lib/fluent/plugin/in_tail.rb#L435

Reply:
> If possible, could you let me know who will call `detach_watcher()` if we removed this line?

`update_watcher` always detach the old watcher in this line:

https://github.com/fluent/fluentd/blob/ce898dd572d5745663abd456486169d772daa8aa/lib/fluent/plugin/in_tail.rb#L532

Reply:
Oh maybe I misunderstood!
You say if we stop unwatching in `update_watcher`, then nobody unwatches its inode and the position remains.
It would be dangerous because a new file coincidently having the same inode would use that position wrongly.

That's right!
The problem of unwatching still remain, so now I think we can't stop unwatching in `update_watcher` in this PR.
(I'm sorry for this. I though we could stop it in this PR, but it was wrong.)

Perhaps I will revert that modification, Let me think about it some more.